crates/sncast/src/helpers/fee.rs

```diff
@@ -82,8 +87,11 @@ impl FeeArgs {
             let mut fee_settings = FeeSettings::try_from(fee_estimate.clone())
                 .expect("Failed to convert FeeEstimate to FeeSettings");
 
-            // If a tip is not provided, set it to 0
-            fee_settings.tip = Some(self.tip.unwrap_or(0));
+            fee_settings.tip = if self.estimate_tip {
+                None // If we leave it as None, the tip will be estimated before sending the transaction
+            } else {
+                Some(self.tip.unwrap_or(0))
+            };
```

Comment:
nit: Maybe we can put it into separate method (e.g. `resolve_tip`) so it will be reused below as well.